lib/spack/spack/environment.py

```diff
@@ -71,6 +77,11 @@
 #: legal first keys in the spack.yaml manifest file
 env_schema_keys = ('spack', 'env')
 
+# Magic names
+user_speclist_name = 'specs'
+def_view_name = 'default'
+def_view_link = 'roots'
```

Comment:
Make this `all` by default -- otherwise regular environments are not going to work for python (and likely other things).  e.g., I think this is a pretty common set of requirements:

```python
spack:
    - py-numpy
    - py-pandas
```

But if you link only `roots` by default, the environment won't even work.  I think `roots` is still only for when you know what you're doing.